src/utils/utils.go

```diff
@@ -228,20 +230,18 @@ func WaitForPredicateWithTimer(ctx context.Context, timeout time.Duration, inter
 	}
 }
 
-func WaitForPredicate(timeout time.Duration, interval time.Duration, predicate func() bool) error {
-	return WaitForPredicateWithContext(context.TODO(), timeout, interval, predicate)
+func WaitForeverWithParam[E any](ctx context.Context, interval time.Duration, predicate func(E) bool, e E) error {
+	return WaitForever(ctx, interval, func() bool {
+		return predicate(e)
+	})
 }
 
-func WaitForPredicateWithContext(ctx context.Context, timeout time.Duration, interval time.Duration, predicate func() bool) error {
-	return WaitForPredicateWithTimer(ctx, timeout, interval, func(timer *time.Timer) bool {
-		return predicate()
-	})
+func WaitForever(ctx context.Context, interval time.Duration, predicate func() bool) error {
```

Comment:
WaitForever is a little bit misleading as for me, we still wait for predicate no?

Reply:
OK.  Will change